scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -704,6 +744,56 @@ impl TomlManifest {
     }
 }
 
+/// Returns the absolute canonical path of the README file for a [`TomlPackage`].
+pub fn readme_for_package(
+    package_root: &Utf8Path,
+    readme: Option<&PathOrBool>,
+) -> Result<Option<Utf8PathBuf>> {
+    match readme {
+        None => abs_canonical_path(
```

Comment:
```suggestion
        Some(PathOrBool(true)) | None => abs_canonical_path(
```

you can do this. note, that you'll have to flatten nested `match` expression in `Some(value)` case, i.e.

```rust
match readme {
    Some(PathOrBool::Bool(true)) | None => smth,
    Some(PathOrBool::Bool(false)) => smth,
    Some(PathOrBool::Path(p)) => smth
}
```

Reply:
Thank you for this reminder!
However, in this case I think it's better to have those match arms separate, instead of combining them to add an if later to separate them (I want to make sure a readme exists if it's set to true, but doesn't have to exist if readme isn't set at all).

Reply:
Yeah, what you have is fine, we can leave this as-is :)